### DISCUSSION TITLE
0.11.0 small regression

### DISCUSSION THREAD

**Discussion by mackoj** [pointfreeco/swift-parsing#265] [General]
Hi,

Thanks for the fixes in the `0.11.0` version it helps a lot but after updating to 0.11.0 a few of my parsers face some small issues.
Where I use `Parse(SomeCustomConvert())` the type of the input changed from.

A few examples:
Before: `public typealias Input = ((String, Int), (String, String, String))` 
After: `public typealias Input = (String, Int, (String, String, String))`

Before: `public typealias Input = ((String, Int, String), (String, Double, Double), ([Double], String, String), (Double?, Double, Double), (Double, Double))`
After: `public typealias Input = (String, Int, String, (String, Double, Double), ([Double], String, String), (Double?, Double, Double), (Double, Double))`

Before: `public typealias Input = ((String, Int?, Int?), String, String)`
After: `public typealias Input = (String, Int?, Int?, String, String)`

For some reason, the first tuple always lost its tupling-ness.

To debug this when I saw `No exact matches in call to initializer` or `Initializer 'init(_:with:)' requires the types X and Y be equivalent` I just comment the `(SomeCustomConvert())` to reveal the type when I add `.map` a the closing `}` of the `Parse`. Then I just redid the Input type to fix it.

// sorry for the multiple edits I did send it by mistake before finishing it

**Comment by stephencelis** [comment]
@mackoj There are some differences expected due to how `buildPartialBlock` accumulates, but we think it's probably a change that needs to happen in order to support many more parsers in a block. Apple's `Regex` builder similarly flattens things, but even more aggressively than we do (we may want to be as aggressive in the future).

The tupling changed slightly for our `Date` benchmark, which you can observe here:

https://github.com/pointfreeco/swift-parsing/commit/645c688e0ba6a06daae8a77073bdb9c3c60ef7b4?w=1#diff-b9c31c5970042d2757410720d6dc581db52c8b951963f09113be33533a1042f0L71-R139

Can you post a full example of the issue, how you expect it to capture output, and then how it does capture output?

**Comment by mackoj** [comment]
This is an example of it.

I expect it to capture content like this `((String, Int), (String, String, String))` but it did it like this `(String, Int, (String, String, String))`.

parser:
![Capture d’écran 2022-12-09 à 16 10 02](https://user-images.githubusercontent.com/661647/206732683-23abd226-7287-4124-b2f1-36c86e06c824.png)

conversion(left after, right before):
![Capture d’écran 2022-12-09 à 16 05 00](https://user-images.githubusercontent.com/661647/206732161-4495f40e-ac97-442f-8616-c017d42f0307.png)

**Comment by mackoj** [comment]
I don't think it's a major issue some of the custom conversions were to go around the limitation of the number of parsers. With the `0.11.0` it's cleaner.
```swift
let failureTestDetails = Parse {
  Peek { "/".utf8 }
  PrefixThrough("\n".utf8).pipe {
    Parse(.case(XcodeBuildLog.testCaseFailureReason)) {
      PrefixUpTo(":".utf8).map(.string) // filename
      ":".utf8
      Int.parser() // line
      ":".utf8
      " error: -[".utf8
      PrefixUpTo(" ".utf8).map(.string) // suite
      " ".utf8
      PrefixUpTo("] : ".utf8).map(.string) // name
      "] : ".utf8
      PrefixUpTo("\n".utf8).map(.string) // failureMessage
      "\n".utf8
    }
  }
}
```
